### PR TITLE
#162413570 Refactor hardcoded status and topics into a constant file

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
- Refactors all hardcoded status and topics strings into constants

#### Description of tasks to be completed
- Find all instances where values for status and topic are hardcoded
- Define the identified values in the `constants.js` file and replace the hardcoded ones with values from the `constants.js` file

#### How should this be manually tested?
- Clone this repo and checkout to a branch named `ch-move-hard-coded-constants-to-constant-file-162413570`.
- Explore the `src` directory and confirm that there is a file named `constants.js` with `STATUS` and `TOPICS` constants defined there.
- Confirm that no hardcode strings for `status` and `topic` in the other files in `src` directory

#### Any background context you want to provide?
Having constants defined in a single file makes it possible to update them when needed rather than having to locate all instances where they have been used to update them. This improves code maintainability.

#### What are the relevant pivotal tracker stories?
[Refactor hardcoded status and topics into a constant file](https://www.pivotaltracker.com/story/show/162413570)

#### Screenshots?
N/A

#### Questions
N/A

### CODE REVIEW

**Comment by megame24** [comment]
Note that you raised this against `master`, I believe it should be raised against `staging`.

File: src/api.js [hunk 1]
```diff
@@ -205,10 +208,10 @@ async function getUserMessages(userId) {
     method: 'GET',
     uri: process.env.API_URL + '/clients/' + userId + '/messages',
     headers: {
-      Authorization: 'Bearer ' + process.env.OAUTH_ACCESS_TOKEN
+      Authorization: 'Bearer ' + process.env.OAUTH_ACCESS_TOKEN,
```
Comment (andela-kakpobome) [inline src/api.js]: Use template interpolation instead of concatenation i.e `Bearer ${process.env.OAUTH_ACCESS_TOKEN}`

File: src/api.js [hunk 2]
```diff
@@ -283,17 +286,16 @@ async function markMediaAsViewed(clientId, mediaId) {
   if (mediaId) {
     const media = await rp({
       method: 'POST',
-      uri:
-        process.env.API_URL +
+      uri: process.env.API_URL +
```
Comment (andela-kakpobome) [inline src/api.js]: As above, use template interpolation here

File: src/api.js [hunk 3]
```diff
@@ -324,10 +326,10 @@ async function getTask(id) {
     method: 'GET',
     uri: process.env.API_URL + '/tasks/' + id,
     headers: {
-      Authorization: 'Bearer ' + process.env.OAUTH_ACCESS_TOKEN
+      Authorization: 'Bearer ' + process.env.OAUTH_ACCESS_TOKEN,
```
Comment (andela-kakpobome) [inline src/api.js]: Here as well, interpolation

File: src/constants.js [hunk 4]
```diff
@@ -1,8 +1,37 @@
...
+};
+
 module.exports = {
   FB: 'fb',
   SMS: 'sms',
   WORKPLAN_URL: 'https://www.helloroo.org/my-tasks',
   INTRO_VIDEO_URL: 'https://youtu.be/WpHtdkKQz8Q',
   DEFAULT_ERR_MESSAGE: 'This message should not be showing up and is an error on our part.',
-  UNAUTHORIZED: 'unauthorized oauth token. Perhaps refresh the bot oauth token?'
+  UNAUTHORIZED: 'unauthorized oauth token. Perhaps refresh the bot oauth token?',
```
Comment (andela-kakpobome) [inline src/constants.js]: Capitalize the `U` in `unauthorized`

**Comment by meshack-mbuvi** [comment]
@andela-kakpobome I have implemented all the changes you had requested on this PR. Kindly confirm.

**Pull request merged**